### PR TITLE
Remove dependency on testing package from libkb

### PR DESCRIPTION
**Pull request by akalin-keybase**
Also add CI check to make sure it doesn't happen again.

### CODE REVIEW

**Comment by patrickxb** [comment]
I'd rather remove those tests for deprovision working without secret store.  And then remove GetSecretStoreNilForTests().

It's impossible to not have secret store now, so (as the comment says) the tests aren't really testing anything.

**Comment by akalin-keybase** [comment]
Cool, that works for me too.

**Comment by akalin-keybase** [comment]
okay @patrickxb PTAL

File: Jenkinsfile [hunk 1]
```diff
@@ -301,6 +301,8 @@ def testGo(prefix) {
                 [ -z "$lint" -o "$lint" = "Lint-free!" ]
             '''
             sh 'test -z $(gofmt -l $(go list ./... | sed -e s/github.com.keybase.client.go.// ))'
+            // Make sure we don't accidentally pull in the testing package.
+            sh '! go list -f \'{{ join .Imports \"\n\" }}\' github.com/keybase/client/go/keybase | grep testing'
```
Comment (patrickxb) [inline Jenkinsfile]: I think you want `join .Deps` here, not `.Imports`.

Comment (akalin-keybase) [reply Jenkinsfile]: You're right! Fixed

**Pull request merged**